core/src/main/scala/geotrellis/store/GeoTrellisRasterSourceLegacy.scala

```diff
@@ -137,23 +235,75 @@ object GeoTrellisRasterSourceLegacy {
+  def readIntersecting(reader: CollectionLayerReader[LayerId], layerId: LayerId, metadata: TileLayerMetadata[_], extent: Extent, bands: Seq[Int]): Option[Raster[MultibandTile]] = {
     val tiles = readTiles(reader, layerId, extent, bands)
     sparseStitch(tiles, extent)
   }
 
-  def read(reader: CollectionLayerReader[LayerId], layerId: LayerId, metadata: TileLayerMetadata[SpatialKey], extent: Extent, bands: Seq[Int]): Option[Raster[MultibandTile]] = {
+  def read(reader: CollectionLayerReader[LayerId], layerId: LayerId, metadata: TileLayerMetadata[_], extent: Extent, bands: Seq[Int]): Option[Raster[MultibandTile]] = {
     val tiles = readTiles(reader, layerId, extent, bands)
     metadata.extent.intersection(extent) flatMap { intersectionExtent =>
       sparseStitch(tiles, intersectionExtent).map(_.crop(intersectionExtent))
     }
   }
+
+  /**
+   * Build function that can build a list of RasterSources
+   * slicing them by time (in case of a temporal raster source).
+   *
+   * @param attributeStore GeoTrellis attribute store
+   * @param dataPath       GeoTrellis catalog DataPath
+   * @param sourceLayers   List of source layers
+   * @param tResolution    Layer temporal resolution, if not provided the function will try to derive it.
+   * @param targetCellType The target cellType
+   */
+  def build(
+    attributeStore: AttributeStore,
+    dataPath: GeoTrellisPath,
+    sourceLayers: Stream[LayerLegacy],
+    tResolution: Option[Long],
+    targetCellType: Option[TargetCellType]
+  ): List[GeoTrellisRasterSourceLegacy] = {
+    val layerId = dataPath.layerId
+    val header = attributeStore.readHeader[LayerHeader](layerId)
+
+    if(header.keyClass.contains("SpatialKey"))
+      new GeoTrellisRasterSourceLegacy(attributeStore, dataPath, sourceLayers, None, targetCellType) :: Nil
+    else {
+      val md = attributeStore.readMetadata[TileLayerMetadata[SpaceTimeKey]](layerId)
+      val keyIndex = attributeStore.readKeyIndex[SpaceTimeKey](layerId)
+      val temporalResolution = tResolution.getOrElse(keyIndex match {
+        case ki if ki.isInstanceOf[ZSpaceTimeKeyIndex]       => ki.asInstanceOf[ZSpaceTimeKeyIndex].temporalResolution
+        case ki if ki.isInstanceOf[HilbertSpaceTimeKeyIndex] => ki.asInstanceOf[ZSpaceTimeKeyIndex].temporalResolution
+        case ki => throw new UnsupportedOperationException(s"Can't derive layer temporal resolution for ${ki.getClass}, try to pass a tResolution parameter explicitly.")
+      })
+
+      md.bounds match {
+        case KeyBounds(minKey, maxKey) =>
+          (minKey.time.toInstant.toEpochMilli to maxKey.time.toInstant.toEpochMilli by temporalResolution).toList.map { time =>
+            new GeoTrellisRasterSourceLegacy(attributeStore, dataPath, sourceLayers, Some(ZonedDateTime.ofInstant(time, ZoneOffset.UTC)), targetCellType)
+          }
```

Comment:
This unfortunately is not correct for temporal layers. There could be time buckets that have no records stored in them and there could be time buckets which have multiple records stored in them. The only way to know which is which is to have use store the used values in metadata. Thinking of a concrete suggestion for this.

Reply:
@echeipesh provide time as an input into this build function? (this can be a part of configuration) 

But what to do if you don't know anything about the temporal layer?

Reply:
Can't really report anything you don't know. The alternative is that you report a date for which there is no data and query comes back empty.